### PR TITLE
Integrate new Terms of Service package

### PR DESCRIPTION
**Pull request by enejb**
Based on https://github.com/Automattic/jetpack/pull/13763 PR
This PR deprecated the Jetpack ::jetpack_tos_agreed(). 
Updates any calls that relied on that method the the new `Terms_of_service->has_agreed();`

Also instantiates any Jetpack tracking as soon as the  sites agrees to the TOS. 

Supersedes https://github.com/Automattic/jetpack/pull/13759 

#### Changes proposed in this Pull Request:
* Implements the new Terms Of Service Class. That is to be added in https://github.com/Automattic/jetpack/pull/13763.

#### Testing instructions:
* Does it break tracking? 
* Does it break the connection? 
* Do you see the jetpack_connection_success again? After the site is connected. 

#### Proposed changelog entry for your changes:
* Integrate new Terms of Service package

### CODE REVIEW

**Comment by kraftbj** [comment]
Instead of having a separate package, what about adding this to the Status package? Whether or not the customer has agreed to the TOS for the site feels, to me, more like a Status of the site than something standalone.

**Comment by kraftbj** [comment]
What about `Status->change->agree_tos()` and `Status->change->revoke_tos()`? The Status package is for determining status or, when applicable, setting?

**Comment by enejb** [comment]
I though about this some more and I think keeping the packages small makes more sense to me. 

The main reason being is that the packages (both Terms_of_Service and Status) becomes more reusable. I can see a plugin needing some so sort of Terms of Service check but doesn't care if we consider the plugin to be in dev mode etc. 

The problem with adding some sort of change object to the Status package is that. 
I am not sure we want to make the status package be responsible for changing state of the application. For example having `Status->is_active()` make sense to me but something like `Status->change->set_active()` (Which would mean handling the connection doesn't. 

I do think it would make sense to add a helper to the status method. 

Would love to hear more thoughts about this ^.

**Comment by tyxla** [comment]
> The Status package is for determining status or, when applicable, setting?

Currently, it's only for determining status, but I don't see why it can't be used for setting too.

File: class.jetpack.php [hunk 1]
```diff
@@ -728,6 +725,23 @@ function( $callable ) {
 		add_action( 'script_loader_tag', array( $this, 'script_add_async' ), 10, 3 );
 	}
 
+	/**
+	 * Runs after all the plugins have loaded but before init.
+	 *
+	 */
+	function after_plugins_loaded() {
+		$terms_of_service = new \Automattic\Jetpack\Terms_Of_Service();
```
Comment (tyxla) [inline class.jetpack.php]: I suggest that we always keep the leading backslash off. It's unnecessary, and when omitted, allows for us to magically mock classes when used in other namespaces.

**Review by tyxla — COMMENTED** [review]
I like the idea about moving this to the status package. Let's go with that if y'all think it makes sense.

**Comment by tyxla** [comment]
We've been chatting with @enejb about moving it to the status package, and I got to the conclusion that this looks a bit better when it's wrapped as a separate package. A `tos` pacakge might be used by packages with different needs than the status package, and it's much less generic than just `status` anyway. I'm fine with both solutions if y'all have strong feelings, but after giving it more thought, I think I don't see any downsides with keeping it a separate package.

**Comment by enejb** [comment]
The plan now is to:
1. Merge https://github.com/Automattic/jetpack/pull/13763 
2. Release the terms of service package. 
3. New PR to update the Tracks Package so that it has a dependecy on the terms of service package. 
4. Release the new tracks package.
5. Rebase this PR agains master. Hopefully the tests will pass then

**Comment by tyxla** [comment]
Just reviewed #13763, happy to review this one later when it's done. Marking as "in progress" in the meantime.

**Comment by enejb** [comment]
Closing this in favor of https://github.com/Automattic/jetpack/pull/13798

**Pull request closed without merging**